udata/features/territories/views.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import csv
+import codecs
+import StringIO
+
+from flask import abort, current_app, request, send_file
+
+from udata import theme
+from udata.models import Dataset
+from udata.i18n import I18nBlueprint
+from udata.utils import multi_to_dict
+from udata.core.storages import references
+
+blueprint = I18nBlueprint('territories', __name__, url_prefix='/territory')
+
+
+@blueprint.route(
+    '/<territory:territory>/dataset/<dataset:dataset>/resource/<resource_id>/',
+    endpoint='territory_dataset_resource')
+def compute_territory_dataset(territory, dataset, resource_id):
+    """
+    Dynamically generate a CSV file about the territory from a national one.
+
+    The file targeted by the resource MUST be downloaded within the
+    `references` folder with the original name prior to call that view.
+
+    The GET paramaters `territory_attr` and `csv_column` are used to
+    determine which attribute MUST match the given column of the CSV.
+    """
+    args = multi_to_dict(request.args)
+    if 'territory_attr' not in args or 'csv_column' not in args:
+        return abort(404)
+    for resource in dataset.resources:
+        if resource.id == resource_id:
+            break
+
+    resource_path = references.path(resource.url.split('/')[-1])
+    match = getattr(territory, args['territory_attr']).encode('utf-8')
```

Comment:
Only if absent from the query, not from territory. If someone query `?territory_attr=wtf` it will raise a 500. It should raise a 400.